client.go

```diff
@@ -69,28 +71,155 @@ type Client struct {
 	// through legitimate channels.
 	dopplegangerAddrs map[string]struct{}
 	badPeerIPs        map[string]struct{}
-	torrents          map[InfoHash]*Torrent
+	torrents          map[InfoHash]*torrent
 
 	acceptLimiter   map[ipStr]int
 	dialRateLimiter *rate.Limiter
 }
 
 type ipStr string
 
+// BadPeerIPs ...
 func (cl *Client) BadPeerIPs() []string {
 	cl.rLock()
 	defer cl.rUnlock()
 	return cl.badPeerIPsLocked()
 }
 
+// Start the specified torrent.
+// Start adds starts up the torrent within the client downloading the missing pieces
+// as needed. if you want to wait until the torrent is completed use Download.
+func (cl *Client) Start(t Torrent) (dl Download, added bool, err error) {
+	return cl.start(t)
+}
+
+// MaybeStart is a convience method that consumes the return types of the torrent
+// creation methods: New, NewFromFile, etc. it is all respects identical to the Start
+// method.
+func (cl *Client) MaybeStart(t Torrent, failed error) (dl Download, added bool, err error) {
+	if failed != nil {
+		return dl, false, failed
+	}
+	return cl.start(t)
+}
+
+// Download the specified torrent.
+// Download differs from Start in that it blocks until the torrent is completely downloaded.
+// the provided ctx is used to timeout the download, but it currently only works with regards
+// to fetching the torrent info.
+func (cl *Client) Download(ctx context.Context, t Torrent) (dl Download, added bool, err error) {
```

Comment:
totally hacky download method. details here can be cleaned up pretty sure the copy is unnecessary.